### ISSUE TITLE
Potential memory leak.

### ISSUE THREAD

**Issue by rachyyyy** [fogleman/Craft#299]
Potential memory leak in[ shell.c](https://github.com/fogleman/Craft/blob/d6888a6e1e54340358ce25b1220f410541360b6b/deps/sqlite/shell.c) line 2357. **The while loop is executed without releasing Zsql will cause a memory leak.** [Doc](https://www.sqlite.org/c3ref/mprintf.html) says "The sqlite3_mprintf() and sqlite3_vmprintf() routines write their results into memory obtained from [sqlite3_malloc64()](https://www.sqlite.org/c3ref/free.html). The strings returned by these two routines should be released by [sqlite3_free()](https://www.sqlite.org/c3ref/free.html)." Using sqlite3_free() to free zSql will prevent the memory leak.

![1662191655452](https://user-images.githubusercontent.com/57169201/188261631-00464de2-df45-45f5-8363-4c027fbc3e7b.png)

**Comment by OctavioAnino** [comment]
zSql is freed in line 2375

**Comment by rachyyyy** [comment]
> zSql is freed in line 2375

For all the mprintf() called in while loop (line 2357 and line 2364), the zSql should be freed after the while loop and before line 2373. Otherwise, there will be a potential memory leak.

**Comment by OctavioAnino** [comment]
You're right, my bad. This is also in contrast with what is seen throughout the rest of the [file](https://github.com/fogleman/Craft/blob/d6888a6e1e54340358ce25b1220f410541360b6b/deps/sqlite/shell.c)-  were every pointer returned by mprintf() is indeed eventually freed. Of course, except the ones in this issue.
